packages/firestore/src/remote/backoff.ts

```diff
@@ -92,18 +94,39 @@ export class ExponentialBackoff {
 
     // First schedule using the current base (which may be 0 and should be
     // honored as such).
-    const delayWithJitterMs = this.currentBaseMs + this.jitterDelayMs();
+    const desiredDelayWithJitterMs = Math.floor(
+      this.currentBaseMs + this.jitterDelayMs()
+    );
+
+    // Some time may have already elapsed so account for that.
+
+    // Math.max(0, ...) to guard against lastAttemptTime being in the future due
+    // to a clock change.
+    const delaySoFarMs = Math.max(0, Date.now() - this.lastAttemptTime);
+
+    // Math.max(0, ...) because the desired backoff may already be elapsed.
```

Comment:
nit: I think the "Some time may have already elapsed" somewhat duplicates this line. Consider axing it.

(also drop "Max.max ... because")